idaes/core/initialization/initializer_base.py

```diff
@@ -0,0 +1,399 @@
+from pyomo.common.config import ConfigBlock, ConfigValue
+
+from idaes.core.util import to_json, from_json, StoreSpec
+from idaes.core.util.exceptions import InitializationError
+from idaes.core.util.model_statistics import degrees_of_freedom
+import idaes.logger as idaeslog
+
+__author__ = "Andrew Lee"
+
+
+class InitializationStatus(Enum):
+    """
+    Enum of expected outputs from Initialization routines.
+    """
+
+    Ok = 1  # Succesfully converged to tolerance
+    Failed = 0  # Run, but failed to converge to tolerance
+    DoF = -1  # Failed due to Degrees of Freedom issue
+    Error = -2  # Exception raised during execution (other than DoF or convergence)
+
+
+StoreState = StoreSpec(
+    data_classes={
+        Var._ComponentDataClass: (("fixed",), None),
+        BooleanVar._ComponentDataClass: (("fixed",), None),
+        Block._ComponentDataClass: (("active",), None),
+        Constraint._ComponentDataClass: (("active",), None),
+    }
+)
+
+
+class InitializerBase:
+    """
+    Base class for Initializer objects.
+
+    This implements a default workflow nad methods for common tasks.
+    Developers should feel free to overload these as necessary.
+    """
+
+    CONFIG = ConfigBlock()
+
+    CONFIG.declare(
+        "constraint_tolerance",
+        ConfigValue(
+            default=1e-5,
+            domain=float,
+            description="Tolerance for checking constraint convergence",
+        ),
+    )
+    CONFIG.declare(
+        "output_level",
+        ConfigValue(
+            default=idaeslog.NOTSET,
+            description="Set output level for logging messages",
+        ),
+    )
+
+    def __init__(self, **kwargs):
+        self.config = self.CONFIG(kwargs)
+
+        self.initial_state = None
+        self.postcheck_summary = {}
+        self.status = None
+
+    def get_logger(self, model):
+        return idaeslog.getInitLogger(model.name, self.config.output_level)
+
+    def initialize(
+        self, model: Block, initial_guesses: dict = None, json_file: str = None
+    ):
+        """
+        Execute full initialization routine.
+
+        Args:
+            model: Pyomo model to be initialized.
+            initial_guesses: dict of initial guesses to load.
+            json_file: file name of json file to load initial guesses from as str.
+
+        Note - can only provide one of initial_guesses or json_file.
+
+        Returns:
+            InitializationStatus Enum
+        """
+        # 1. Get current model state
+        self.get_current_state(model)
+
+        # 2. Load initial guesses
+        self.load_initial_guesses(
+            model, initial_guesses=initial_guesses, json_file=json_file
+        )
+
+        # 3. Fix states to make square
+        self.fix_initialization_states(model)
+
+        # 4. Prechecks
+        self.precheck(model)
+
+        # 5. try: Call specified initialization routine
+        try:
+            results = self.initialization_routine(model)
```

Comment:
Why do you ignore exceptions here?  Or is it just that you haven't gotten to it?

Reply:
In part hadn't got to it, but also letting the postcheck handle any errors that need to be raised.